// @awesome.me/webawesome@3.0.0-beta.1/dist/webawesome.ssr-loader.js
import "./chunks/chunk.37ZGXZDM.js";
import {
  allDefined,
  discover,
  preventTurboFouce,
  serialize,
  startLoader,
  stopLoader
} from "./chunks/chunk.2QJYSVFW.js";
import {
  registerTranslation
} from "./chunks/chunk.HMQGKY4F.js";
import "./chunks/chunk.V7FECYHX.js";
import {
  getAnimationNames,
  getEasingNames
} from "./chunks/chunk.TJOP7HQP.js";
import {
  getDefaultIconFamily,
  registerIconLibrary,
  setDefaultIconFamily,
  unregisterIconLibrary
} from "./chunks/chunk.OUUMRN6Q.js";
import {
  getBasePath,
  getKitCode,
  setBasePath,
  setKitCode
} from "./chunks/chunk.HNHPBE3O.js";
import "./chunks/chunk.UPHXFDHQ.js";
import "./chunks/chunk.RBKXYBVR.js";

// src/webawesome.ssr-loader.ts
import "@lit-labs/ssr-client/lit-element-hydrate-support.js";
export {
  allDefined,
  discover,
  getAnimationNames,
  getBasePath,
  getDefaultIconFamily,
  getEasingNames,
  getKitCode,
  preventTurboFouce,
  registerIconLibrary,
  registerTranslation,
  serialize,
  setBasePath,
  setDefaultIconFamily,
  setKitCode,
  startLoader,
  stopLoader,
  unregisterIconLibrary
};
